lib/ecto/query/builder/select.ex

```diff
@@ -98,16 +99,19 @@ defmodule Ecto.Query.Builder.Select do
   end
 
   # map/struct(var, [:foo, :bar])
-  defp escape({tag, _, [{var, _, context}, fields]}, {params, take}, vars, env)
+  defp escape({tag, _, [{var, _, context}, fields]}, {params, {take, subqueries}}, vars, env)
        when tag in [:map, :struct] and is_atom(var) and is_atom(context) do
     taken = escape_fields(fields, tag, env)
     expr = Builder.escape_var!(var, vars)
     take = add_take(take, Builder.find_var!(var, vars), {tag, taken})
-    {expr, {params, take}}
+    {expr, {params, {take, subqueries}}}
   end
 
-  defp escape(expr, params_take, vars, env) do
-    Builder.escape(expr, :any, params_take, vars, {env, &escape_expansion/5})
+  defp escape(expr, {_params, {take, _subqueries}} = params_take, vars, env) do
+    case Builder.escape(expr, :any, params_take, vars, {env, &escape_expansion/5}) do
+      {expr, {params, {take, subqueries}}} -> {expr, {params, {take, subqueries}}}
+      {expr, {params, subqueries}} -> {expr, {params, {take, subqueries}}}
```

Comment:
If this is missing the `take`, it means a bug somewhere. This clause should not exist.

Reply:
Can you please rename `params_take` in this file to `params_acc`, as it is more than just take?